runtime/array.cu

```diff
@@ -191,59 +191,52 @@ __device__ void loadGlobalToLocal(
       loadGenericVolatile<scalar_t, vec_size, false, is_volatile>(to, from);
       break;
     case 8: {
+      uint2& data = *reinterpret_cast<uint2*>(to);
+      const auto* from_address = reinterpret_cast<
+          const typename MaybeVolatile<uint2, is_volatile>::type*>(from);
       if (is_volatile) {
-        uint2& data = *reinterpret_cast<uint2*>(to);
+        data = *from_address;
+#if 0
```

Comment:
I'm not sure which conditions triggers this so I can make sure `ld.volatile` was produced.

Reply:
I don't think none of the current scheduler would produce this, but probably something this would result in using volatile:

```
auto tv0 = makeSymbolicTensor(1);
fusion.addInput(tv0);
auto tv1 = set(tv0);
auto tv2 = set(tv1);
auto tv3 = set(tv2);
fusion.addOutput(tv3);

tv1->axis(0)->parallelize(ParallelType::BIDx);
tv1->setMemoryType(MemoryType::Global);
tv2->split(0, 2);
tv2->axis(0)->parallelize(ParallelType::BIDx);
tv2->axis(1)->parallelize(ParallelType::Vectorize);

fusion.printKernel();
```

Notice that `tv1` and `tv2` are parallelized differently with `BIDx`, so the copy op, `tv2 = tv1`, needs to have `tv1` in some shared memory that all thread blocks can access, i.e., global memory. Furthermore, for the consistency of `tv1` data on global memory, there will be a global sync and the store and load from `tv1` should be done with `volatile`. Since `tv2` is also vectorized, I believe this should result in using this code path.